crates/ruff_linter/src/rules/pylint/rules/nested_min_max.rs

```diff
@@ -20,6 +20,19 @@ pub(crate) enum MinMax {
 /// Nested `min` and `max` calls can be flattened into a single call to improve
 /// readability.
 ///
+/// ## Fix safety
+/// This fix is always unsafe and may change the program's behavior when expanding
+/// `min()` or `max()` calls. The underlying dunder methods (`__lt__` for `min` and
+/// `__gt__` for `max`) are not associative, particularly when dealing with special
+/// floating-point values like `NaN` (Not a Number).
```

Comment:
```suggestion
/// This fix is always unsafe and may change the program's behavior for
/// types without full equivalence relations, such as float comparisons involving `NaN`.
```

I _think_ associativity is the correct property, but this is another possibility that I borrowed from the Rust [`PartialEq`](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html) docs. I think we could also phrase it as "types without a total ordering" or something. Maybe associative is close enough to stick with that :sweat_smile: